### PR TITLE
Update nix to 0.23

### PR DESCRIPTION
**Pull request by taiki-e**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [Unreleased] - ReleaseDate
+
+### Changed
+
+- Updated Nix to 0.23.
```
Comment (asomers) [inline CHANGELOG.md]: You should mention the MSRV change too.

**Review by asomers — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (taiki-e) [reply CHANGELOG.md]:
```suggestion
- Updated Nix to 0.23.  This change raises MSRV from 1.41 to 1.46.
```

**Comment by taiki-e** [comment]
Updated to mention MSRV change in changelog.

**Comment by asomers** [comment]
Thanks!

**Pull request merged**